src/java.base/share/classes/java/security/PEMDecoder.java

```diff
@@ -69,44 +70,46 @@
+ * <p> For {@code PublicKey} and {@code PrivateKey} types, an algorithm-specific
+ * subclass is returned if the algorithm is supported. For example an
  * ECPublicKey and ECPrivateKey for Elliptic Curve keys.
  *
  * <p> If the PEM type does not have a corresponding class,
  * {@code decode(String)} and {@code decode(InputStream)} will return a
- * {@link PEMRecord}.
+ * {@link PEM}.
  *
- * <p> The {@linkplain #decode(String, Class)} and
- * {@linkplain #decode(InputStream, Class)} methods take a class parameter
- * which determines the type of {@code DEREncodable} that is returned. These
+ * <p> The {@link #decode(String, Class)} and
+ * {@link #decode(InputStream, Class)} methods take a class parameter
+ * to specify the returned {@code DEREncodable} type. These
  * methods are useful when extracting or changing the return class.
- * For example, if the PEM contains both public and private keys, the
- * class parameter can specify which to return. Use
- * {@code PrivateKey.class} to return only the private key.
- * If the class parameter is set to {@code X509EncodedKeySpec.class}, the
- * public key will be returned in that format.  Any type of PEM data can be
- * decoded into a {@code PEMRecord} by specifying {@code PEMRecord.class}.
+ * For example, if the PEM contains both a public and private keys, specifying
+ * {@code PrivateKey.class} returns only the private key.
+ * If the class parameter specifies {@code X509EncodedKeySpec.class}, the
+ * public key encoding is returned in that class.  Any type of PEM data can be
+ * decoded into a {@code PEM} object by specifying {@code PEM.class}.
  * If the class parameter doesn't match the PEM content, a
  * {@linkplain ClassCastException} will be thrown.
  *
  * <p> A new {@code PEMDecoder} instance is created when configured
- * with {@linkplain #withFactory(Provider)} and/or
- * {@linkplain #withDecryption(char[])}. {@linkplain #withFactory(Provider)}
- * configures the decoder to use only {@linkplain KeyFactory} and
- * {@linkplain CertificateFactory} instances from the given {@code Provider}.
- * {@linkplain #withDecryption(char[])} configures the decoder to decrypt all
+ * with {@link #withFactory(Provider)} and/or
+ * {@link #withDecryption(char[])}. {@link #withFactory(Provider)}
+ * restricts decoding to {@link KeyFactory} and
+ * {@link CertificateFactory} instances from the specified {@code Provider}.
+ * {@link #withDecryption(char[])} configures the decoder to process
  * encrypted private key PEM data using the given password.
- * Configuring an instance for decryption does not prevent decoding with
- * unencrypted PEM. Any encrypted PEM that fails decryption
- * will throw a {@link RuntimeException}. When an encrypted private key PEM is
- * used with a decoder not configured for decryption, an
- * {@link EncryptedPrivateKeyInfo} object is returned.
+ * If decryption fails, a {@link RuntimeException} is thrown.
```

Comment:
Be more specific - `IllegalArgumentException` is specified by the decode methods.